src/nr_openai_observability/patchers/openai_streaming.py

```diff
@@ -34,9 +34,12 @@ def wrap_stream_generator(stream_gen):
                 trace.add_custom_attribute("completion_id", completion_id)
                 handle_start_completion(kwargs, completion_id)
                 for chunk in stream_gen:
-                    content += chunk.choices[0].delta.get("content", "")
-                    if hasattr(chunk.choices[0].delta, "role"):
-                        role = chunk.choices[0].delta.role
+                    if len(chunk.choices) > 0:
+                        content += chunk.choices[0].delta.get("content", "") or ""
+                        delta_role = getattr(chunk.choices[0].delta, "role", None)
```

Comment:
It seems like the content or role can both be `None` even after a previous chunk had a role. Only track the role when there's a value here